### PR TITLE
s/preact/react/. Strip non-prod code from react-dates bundle. Import normally

### PR DESCRIPTION
**Pull request by cvializ**
This decreases the amp-date-picker dist binary size and uses more normal import mechanisms.

### CODE REVIEW

File: extensions/amp-date-picker/0.1/amp-date-picker.js [hunk 1]
```diff
@@ -230,7 +229,7 @@ class AmpDatePicker extends AMP.BaseElement {
    * @return {?Promise<!JsonObject|!Array<JsonObject>>}
    */
   fetchSrcTemplates_() {
-    if (assertHttpsUrl(this.element.getAttribute('src'), this.element)) {
```
Comment (cvializ) [inline extensions/amp-date-picker/0.1/amp-date-picker.js]: `assertHttpsUrl` is already called in `fetchedBatchedJsonFor`, so it was redundant here.

File: extensions/amp-date-picker/0.1/date-range-picker.js [hunk 2]
```diff
@@ -169,21 +169,6 @@ function createDateRangePickerBase() {
      * @param {?string} focusedInput
      */
     onFocusChange(focusedInput) {
-      // TODO(cvializ): there is a bug in the react-dates library that
```
Comment (cvializ) [inline extensions/amp-date-picker/0.1/date-range-picker.js]: This bug only happened when using `preact-compat` with `react-dates`, so this workaround code isn't needed any more.

**Comment by choumx** [comment]
> decreases the amp-date-picker dist binary size

Surprising. I thought preact's value prop was significantly reduced size?

**Comment by cvializ** [comment]
No that is still true, but the way the PR decreases the size is using `envify` and `uglify` to strip blocks where `if (process.env.NODE_ENV === 'development') {}`. The easiest way to fix the `onFocusChange` bug in `DateRangePicker` was to use `react`. I will check again in the near-future to see if the size difference is enough of an improvement to justify trying to fix the `onFocusChange` bug.

File: extensions/amp-date-picker/0.1/amp-date-picker.css [hunk 3]
```diff
@@ -15,3 +15,7 @@
  */
 
 @import '../../../node_modules/react-dates/lib/css/_datepicker.css';
+
+.DateInput_input {
+  font-size: 100%;
```
Comment (cvializ) [inline extensions/amp-date-picker/0.1/amp-date-picker.css]: Prevents iOS from zooming in when tapping on the hidden input. The default was `11px`. Anything less than `16px` triggers auto-zooming on focus.

Comment (aghassemi) [reply extensions/amp-date-picker/0.1/amp-date-picker.css]: wow, good to know!

**Pull request merged**